src/api/mod.rs

```diff
@@ -745,38 +790,83 @@ impl<T: Pixel> Context<T> {
   }
 
-  /// Ask how many bytes of the stats file are needed before the next frame
-  /// of the second pass in a two-pass encode can be encoded. This is a lower
-  /// bound (more might be required), but if 0 is returned, then encoding can
-  /// proceed. This is just a hint to the application, and does not need to
-  /// be called for encoding the second pass to work, so long as the
-  /// application continues to provide more data to twopass_in() in a loop
-  /// until twopass_in() returns 0.
+  /// Returns the number of bytes of the stats file needed before the next
+  /// frame of the second pass in a two-pass encode can be encoded.
+  ///
+  /// This is a lower bound (more might be required), but if `0` is returned,
+  /// then encoding can proceed. This is just a hint to the application, and
+  /// does not need to be called for encoding the second pass to work, so long
+  /// as the application continues to provide more data to [`twopass_in`] in a
+  /// loop until [`twopass_in`] returns `0`.
+  ///
+  /// [`twopass_in`]: #method.twopass_in
+  #[inline]
   pub fn twopass_bytes_needed(&mut self) -> usize {
     self.inner.rc_state.twopass_in(None).unwrap_or(0)
   }
 
-  /// Provide stats data produced in the first pass of a two-pass encode to the
-  /// second pass. On success this returns the number of bytes of that data
-  /// which were consumed. When encoding the second pass of a two-pass encode,
-  /// this should be called repeatedly in a loop before every call to
-  /// receive_packet() (including the very first one) until no bytes are
-  /// consumed, or until twopass_bytes_needed() returns 0.
+  /// Provides the stats data produced in the first pass of a two-pass encode
+  /// to the second pass.
+  ///
+  /// On success this returns the number of bytes of the data which were
+  /// consumed. When encoding the second pass of a two-pass encode, this should
+  /// be called repeatedly in a loop before every call to [`receive_packet`]
+  /// (including the very first one) until no bytes are consumed, or until
+  /// [`twopass_bytes_needed`] returns `0`.
+  ///
+  /// [`receive_packet`]: #method.receive_packet
+  /// [`twopass_bytes_needed`]: #method.twopass_bytes_needed
+  #[inline]
   pub fn twopass_in(&mut self, buf: &[u8]) -> Result<usize, EncoderStatus> {
     self.inner.rc_state.twopass_in(Some(buf)).or(Err(EncoderStatus::Failure))
   }
 
+  /// Returns the encoded data.
+  ///
+  /// # Examples
+  ///
+  /// ``` no_run
+  /// use rav1e::prelude::*;
+  /// #
+  /// # fn main() -> Result<(), EncoderStatus> {
+  /// let cfg = Config::default();
+  /// let mut ctx: Context<u8> = cfg.new_context();
+  /// let frame = ctx.new_frame();
+  ///
+  /// ctx.send_frame(frame)?;
+  /// ctx.flush();
+  ///
+  /// loop {
+  ///     match ctx.receive_packet() {
+  ///         Ok(packet) => { /* ... */ },
+  ///         Err(EncoderStatus::Encoded) => (),
+  ///         Err(EncoderStatus::LimitReached) => break,
```

Comment:
This is missing a bunch of possible stauses.

Reply:
As far as I can tell, these two are the "success" statuses, and all other ones are "error", so I return them (similarly to how you use `?` in the examples to return errors).

Reply:
I wouldn't call `NeedMoreData` an error; it's a normal part of the encoding flow.

Reply:
Not after a `flush`, which I did in this example.

Reply:
You're right, but that's not a particularily good example then.

Reply:
I should add another example which demonstrates `NeedMoreData`. This one can stay because is probably the simplest way you can use `receive_packet()`, and also it's ok for something like image encoding.

Reply:
That sounds fine to me.